### PR TITLE
APP-1590 Define all interfaces and factories required to implement 'symphony-app-authentication-filter' library

### PR DESCRIPTION
**Pull request by robson-sanchez**
- POM files
- Interfaces and factories
- Unit tests

Reference: https://perzoinc.atlassian.net/wiki/spaces/PLAT/pages/185794629/Tech+Design+Third-party+application+authentication+filter

### CODE REVIEW

File: symphony-app-authentication/symphony-app-authentication-filter/src/test/java/org/symphonyoss/symphony/apps/authentication/factories/ServicesInfoProviderFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  private static final ServicesInfoProvider MOCK_PROVIDER =
+      new MockServicesInfoProvider(MOCK_POD_URL, MOCK_SESSION_AUTH_URL);
+
+  private ServicesInfoProviderFactory factory = ServicesInfoProviderFactory.getInstance();
+
+  @Test
+  public void testNullComponent() {
+    try {
+      factory.setComponent(null);
+      fail();
```
Comment (crepache) [inline symphony-app-authentication/symphony-app-authentication-filter/src/test/java/org/symphonyoss/symphony/apps/authentication/factories/ServicesInfoProviderFactoryTest.java:30]: Is it necessary? On method setComponent, if passed null to method setComponent then should return the exception.

File: symphony-app-authentication/symphony-app-authentication-filter/src/test/java/org/symphonyoss/symphony/apps/authentication/factories/ServicesInfoProviderFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    } catch (IllegalArgumentException e) {
+      assertEquals("Invalid provider implementation. It mustn't be null", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testFactory() {
+    try {
+      factory.getComponent();
+      fail();
```
Comment (crepache) [inline symphony-app-authentication/symphony-app-authentication-filter/src/test/java/org/symphonyoss/symphony/apps/authentication/factories/ServicesInfoProviderFactoryTest.java:40]: The same comment above?

File: symphony-app-authentication/pom.xml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    <scm>
+        <connection>scm:git:git@github.com:symphonyoss/GroupID.git</connection>
+        <developerConnection>scm:git:git@github.com:symphonyoss/GroupID.git</developerConnection>
+        <url>https://github.com/symphonyoss/GroupID</url>
+    </scm>
+
+    <properties>
+        <!-- JDK -->
+        <maven.compiler.source>1.7</maven.compiler.source>
```
Comment (crepache) [inline symphony-app-authentication/pom.xml:103]: Is it possible use Java 1.8?

**Review by crepache — CHANGES_REQUESTED** [review]

File: symphony-app-authentication/pom.xml (hunk 3 above)
Comment (robson-sanchez) [reply symphony-app-authentication/pom.xml:103]: No, it isn't in this moment. This library will be used by Integration Bridge that requires Java 1.7

File: symphony-app-authentication/symphony-app-authentication-filter/src/test/java/org/symphonyoss/symphony/apps/authentication/factories/ServicesInfoProviderFactoryTest.java (hunk 1 above)
Comment (robson-sanchez) [reply symphony-app-authentication/symphony-app-authentication-filter/src/test/java/org/symphonyoss/symphony/apps/authentication/factories/ServicesInfoProviderFactoryTest.java:30]: I'm following this approach to make sure the exception has been thrown by the component and check the exception message.

**Review by crepache — APPROVED** [review]
LGTM

**Pull request merged**